data-prepper-plugins/lambda-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/lambda/LambdaSinkService.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.lambda;
+
+import io.micrometer.core.instrument.Counter;
+import org.opensearch.dataprepper.model.codec.OutputCodec;
+import org.opensearch.dataprepper.model.configuration.PluginSetting;
+import org.opensearch.dataprepper.model.event.EventHandle;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.sink.OutputCodecContext;
+import org.opensearch.dataprepper.model.types.ByteCount;
+import org.opensearch.dataprepper.plugins.sink.lambda.accumlator.Buffer;
+import org.opensearch.dataprepper.plugins.sink.lambda.accumlator.BufferFactory;
+import org.opensearch.dataprepper.plugins.sink.lambda.config.BatchOptions;
+import org.opensearch.dataprepper.aws.api.AwsCredentialsSupplier;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.sink.lambda.codec.JsonCodec;
+import org.opensearch.dataprepper.plugins.sink.lambda.dlq.DlqPushHandler;
+import org.opensearch.dataprepper.plugins.sink.lambda.dlq.LambdaSinkFailedDlqData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.lambda.LambdaClient;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class LambdaSinkService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LambdaSinkService.class);
+    public static final String NUMBER_OF_RECORDS_FLUSHED_TO_LAMBDA_SUCCESS = "lambdaSinkObjectsEventsSucceeded";
+    public static final String NUMBER_OF_RECORDS_FLUSHED_TO_LAMBDA_FAILED = "lambdaSinkObjectsEventsFailed";
+    private final PluginSetting pluginSetting;
+    private final Lock reentrantLock;
+    private final LambdaSinkConfig lambdaSinkConfig;
+    private LambdaClient lambdaClient;
+    private final String functionName;
+    private static Boolean isSync;
+    private int maxEvents = 0;
+    private ByteCount maxBytes = null;
+    private Duration maxCollectionDuration = null;
+    private int maxRetries = 0;
+    private final Counter numberOfRecordsSuccessCounter;
+    private final Counter numberOfRecordsFailedCounter;
+    private final String SYNC_INVOCATION_TYPE = "RequestResponse";
```

Comment:
I think we discussed that sync doesn't make sense in case of lambda as sink. So, invocation type is not needed.